// ant-design-vue/lib/vc-input-number/src/InputHandler.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = void 0;

var _vue = require("vue");

var _vueTypes = _interopRequireDefault(require("../../_util/vue-types"));

var _vcMFeedback = _interopRequireDefault(require("../../vc-m-feedback"));

var _propsUtil = require("../../_util/props-util");

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function _isSlot(s) {
  return typeof s === 'function' || Object.prototype.toString.call(s) === '[object Object]' && !(0, _vue.isVNode)(s);
}

var InputHandler = {
  name: 'InputHandler',
  inheritAttrs: false,
  props: {
    prefixCls: _vueTypes.default.string,
    disabled: _vueTypes.default.looseBool
  },
  render: function render() {
    var _slot;

    var _this$$props = this.$props,
        prefixCls = _this$$props.prefixCls,
        disabled = _this$$props.disabled;
    var touchableProps = {
      disabled: disabled,
      activeClassName: "".concat(prefixCls, "-handler-active")
    };
    return (0, _vue.createVNode)(_vcMFeedback.default, touchableProps, _isSlot(_slot = (0, _vue.createVNode)("span", this.$attrs, [(0, _propsUtil.getSlot)(this)])) ? _slot : {
      default: function _default() {
        return [_slot];
      }
    });
  }
};
var _default2 = InputHandler;
exports.default = _default2;
